maestrowf/datastructures/core/executiongraph.py

```diff
@@ -906,12 +920,32 @@ def execute_ready_steps(self):
             # Now, we need to take the min of the length of the queue and the
             # computed number of slots. We could have free slots, but have less
             # in the queue.
-            _available = min(_available, len(self.ready_steps))
+            # _available = min(_available, len(self.ready_steps))  # Kind of inconsistent if it's affected by queue size sometimes
```

Comment:
I'm not 100% sure what you mean by inconsistent here. This prevents the need for the conditional below (at least that's what it's intent was when it was originally coded. It means not checking for the `-1` value explicitly also.

Reply:
Problem with this is PriorityQueue's are like standard Queue's and do not implement a length like the dequeue that used to live here.  The size is also only approximate (unsure if this is really only the case in multithreading cases or not)

Reply:
Also this whole block can now be removed as we don't need this check -> the if queue.empty() test on the while loop accomplishes the same behavior and exits when there are no more steps are available